sklearn/model_selection/_search.py

```diff
@@ -1484,3 +1488,504 @@ def _run_search(self, evaluate_candidates):
+            refit_end_time = time.time()
+            self.refit_time_ = refit_end_time - refit_start_time
+
+        # Store the only scorer not as a dict for single metric evaluation
+        self.scorer_ = scorers if self.multimetric_ else scorers['score']
+
+        self.cv_results_ = results
+        self.n_splits_ = n_splits
+
+        return self
+
+    def _form_next_generation(self, results):
+        print(max(results["mean_test_score"]),
+              np.mean(results["mean_test_score"]))
+        self._move_previous_best_to_the_beginning(results)
+        self._two_sex_reproduction()
+        self._unisex_reproduction()
+        self._totally_new = self._form_random_genome(self._n_totally_new)
+
+    def _move_previous_best_to_the_beginning(self, results):
+        best_indices = np.argpartition(
+            results["mean_test_score"],
+            self._n_previous_best)[-self._n_previous_best:]
+        if len(self._totally_new) > self._n_totally_new:  # initial run
+            self._previous_best = np.take(self._totally_new, best_indices)
+        else:
+            self._previous_best = np.take(self.current_generation, best_indices)
+
+    def _two_sex_reproduction(self):
+        self._two_sex_reproduced = []
+        for _ in range(self._n_two_sex_reproduced):
+            parent1, parent2 = self._rng.choice(self._previous_best, 2)
+            child = copy(parent1)
+            bit_mask = self._rng.random(len(child))
+            for (name, value), is_parent2_inheritance_proba in \
+                    zip(parent2.items(), bit_mask):
+                if is_parent2_inheritance_proba > 0.5:
+                    child[name] = value
+            self._two_sex_reproduced.append(child)
+
+    def _unisex_reproduction(self):
+        self._unisex_reproduced = []
+        for _ in range(self._n_unisex_reproduced):
+            parent = self._rng.choice(self._previous_best, 1)[0]
+            child = copy(parent)
+            bit_mask = self._rng.random(len(child))
+            signs = self._rng.choice([-1, 1], len(child), replace=True)
+            for (name, value), mutate_proba, sign in \
+                    zip(parent.items(), bit_mask, signs):
+                if mutate_proba > self.mutation_proba:
+                    if type(self.params[name]) == dict:
+                        self._mutate_float(child, name, sign, value)
+                    else:
+                        self._mutate_choise(child, name)
+            self._unisex_reproduced.append(child)
+
+    def _mutate_float(self, child, name, sign, value):
+        delta = self._rng.uniform(self.mutation_value_min,
+                                  self.mutation_value_max)
+        delta *= 1.0 - self._current_iteration / self.n_iter
+        delta *= sign
+        mutated_value = value + delta * value
+        if self.params[name]["min_"] <= mutated_value \
+                <= self.params[name]["max_"]:
+            child[name] = mutated_value
```

Comment:
I have the feeling that this way of mutating numeric hyperparameters is suboptimal because it does not respect the prior distribution of the parameter search space.

For instance, for regularization parameters (and many other hyperparameters), only positive numbers make sense and there natural scaling would be on a exponential scale from 1e-6 to 1e6 for instance. A natural way to sample from this kind of space would be to use a loguniform distribution (which is still not part of sklearn / scipy at this point in time, see #11232).